### PR TITLE
Bump .NET to 6.0.100-preview.2.21114.3.

### PR DESCRIPTION
**Pull request by rolfbjarne**
This required a few other changes, because the linker changed significantly.

~This is still a draft because some of the changes here aren't final (and there's some debug spew as well), I just want a CI run to find everything that breaks.~

### CODE REVIEW

File: tools/dotnet-linker/Steps/CollectAssembliesStep.cs [hunk 1]
```diff
@@ -1,16 +1,41 @@
...
 		{
 			base.TryProcessAssembly (assembly);
 
 			Configuration.Assemblies.Add (assembly);
 		}
+
+		protected override void TryEndProcess ()
+		{
+			foreach (var asm in Configuration.Assemblies)
+				System.Console.WriteLine ($"Collected assembly: {asm.FullName}");
```
Comment (spouliot) [inline tools/dotnet-linker/Steps/CollectAssembliesStep.cs:35]: no need for `System.` prefix
should be under a verbosity check

File: tools/dotnet-linker/Steps/CollectAssembliesStep.cs [hunk 2]
```diff
@@ -1,16 +1,41 @@
...
 
+		protected override void TryProcess ()
+		{
+			base.TryProcess ();
+
+			var tryResolve = Configuration.Context.GetType ().GetMethod ("TryResolve");
+			foreach (var asm in Configuration.ManagedAssembliesToLink) {
+				var ad = (AssemblyDefinition) tryResolve.Invoke (Configuration.Context, new object [] { Path.GetFileNameWithoutExtension (asm) });
+				if (ad != null) {
+					Console.WriteLine ($"Loaded {asm} => {ad.FullName} => {ad.MainModule.FileName}");
```
Comment (spouliot) [inline tools/dotnet-linker/Steps/CollectAssembliesStep.cs:18]: should be under a verbosity check

File: tools/dotnet-linker/Steps/CollectAssembliesStep.cs [hunk 3]
```diff
@@ -1,16 +1,41 @@
...
+				} else {
+					Console.WriteLine ($"Failed to load {asm}");
```
Comment (spouliot) [inline tools/dotnet-linker/Steps/CollectAssembliesStep.cs:20]: should be reported as an error ? or warning

**Review by mandel-macaque — APPROVED** [review]

File: tools/dotnet-linker/Steps/CollectAssembliesStep.cs (hunk 1 above)
Comment (rolfbjarne) [reply tools/dotnet-linker/Steps/CollectAssembliesStep.cs]: Fixed.

File: tools/dotnet-linker/Steps/CollectAssembliesStep.cs (hunk 2 above)
Comment (rolfbjarne) [reply tools/dotnet-linker/Steps/CollectAssembliesStep.cs]: Fixed.

File: tools/dotnet-linker/Steps/CollectAssembliesStep.cs (hunk 3 above)
Comment (rolfbjarne) [reply tools/dotnet-linker/Steps/CollectAssembliesStep.cs]: Completely removed.

**Review by dalexsoto — APPROVED** [review]

**Review by spouliot — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Review by emaf — APPROVED** [review]

**Pull request merged**